scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,557 @@
+        self.kwargs['method'] = 'L-BFGS-B'
+        self.kwargs['options'] = {
+            'maxiter': ls_max_iter,
+        }
+        self.kwargs['bounds'] = list(zip(self.lower, self.upper))
+
+    def local_search(self, x):
+        x_tmp = np.copy(x)
+        fun_temp = self.func(x, *self.fun_args)
+        mres = self.minimizer(self.func, x, *self.fun_args, **self.kwargs)
+        self.nb_fun_call += mres.nfev 
+        # Check if is valid value
+        is_finite = np.all(np.isfinite(mres.x)) and np.isfinite(mres.fun)
+        in_bounds = np.all(mres.x >= self.lower) and np.all(
+            mres.x <= self.upper)
+        is_valid = is_finite and in_bounds
+
+        # Use the new point only if it is valid and return a better results
+        if is_valid and mres.fun < fun_temp:
+            return mres.fun, mres.x
+        else:
+            return fun_temp, x_tmp 
+
+
+def simulated_annealing(func, x0, bounds, maxiter=1000, local_search_options={},
+        initial_temp=5230., visit=2.62, accept=-5.0, maxfun=1e7, seed=None,
+        no_local_search=False):
+    """
+    Find the global minimum of a function using the Simulated Dual Annealing
+    algorithm.
+
+    Parameters
+    ----------
+    func : callable
+        The objective function to be minimized.  Must be in the form
+        ``f(x, *args)``, where ``x`` is the argument in the form of a 1-D array
+        and ``args`` is a  tuple of any additional fixed parameters needed to
+        completely specify the function.
+    x0 : ndarray, shape(n,)
+        A single initial starting point coordinates. If ``None`` is provided,
+        initial coordinates are automatically generated.
+    bounds : sequence, shape (n, 2) 
+        Bounds for variables.  ``(min, max)`` pairs for each element in ``x``,
+        defining bounds for the objective function parameter.
+    maxiter : int, optional
+        The maximum number of global search iterations. Default value is 1000.
+    local_search_options : dict, optional
+        Extra keyword arguments to be passed to the local minimizer
+            ``scipy.optimize.minimize()`` Some important options could be:
+            method : str
+                The minimization method (e.g. ``"L-BFGS-B"``)
+            args : tuple
+                Extra arguments passed to the objective function (``func``) and
+                its derivatives (Jacobian, Hessian).
+    initial_temp : float, optional
+        The initial temperature, use higher values to facilitates a wider
+        search of the energy landscape, allowing simulated_annealing to escape
+        local minima that it is trapped in. Default value is 5230.
+    visit : float, optional
+        Parameter for visiting distribution. Higher values give the visiting
+        distribution a heavier tail, this makes the algorithm jump to a more
+        distant region. The value range is (0, 3]. Default value is 2.62. 
+    accept : float, optional
```

Comment:
what is a sensible range for `accept`? Are there definite bounds that shouldn't be exceeded?